setup.py

```diff
@@ -1,8 +1,16 @@
 from setuptools import find_packages, setup
+from distutils.util import convert_path
+
+# Imports __version__, reference: https://stackoverflow.com/a/24517154/2220152
+ns = {}
+ver_path = convert_path('nbgitpuller/version.py')
+with open(ver_path) as ver_file:
+    exec(ver_file.read(), ns)
+__version__ = ns['__version__']
```

Comment:
I did that for many years. You may want to take a look at [pbr](https://docs.openstack.org/pbr/latest/), it removes all these hacks. Plus it can automatically derives the version from the git tag. Pretty Cool

Reply:
Thank you for introducing this to me @gsemet! It didn't feel good resorting to this method even though I prefer doing it like this more than either duplicating the version number or solving the `index.js` invalidation issue with greater complexity.

I wonder if pbr is used by any other Jupyter-project.

Reply:
I've used PBR in some small projects and it's definitely nice, but import/exec like this is lighter weight and makes the required standard `pkg.__version__` string available, unlike pbr (or anything that I'm aware of) without adding pbr as a *runtime* dependency.

What I would love is something similar to pbr, but generated a *static* `pkg/_version.py` that defined `__version__`, `version_info` at build time, rather than relying on importing pbr.

Getting the version from VCS also means that it breaks on things like git-archive, where git info is unavailable, e.g. `pip install https://github.com/data-8/nbgitpuller/archive/master.zip` won't work for a pbr project, so it's less robust than internally stored versions.

Reply:
i would agree with you :) I would like to have part of pbr, not necessarily everything